File: day-3/challenge1.py
with open('input') as file:
  rucksacks = file.read().splitlines()
  points = 0

  for rucksack in rucksacks:
    component1 = rucksack[:len(rucksack)//2]
    component2 = rucksack[len(rucksack)//2:]

    item = set(component1).intersection(component2).pop()
    points += ord(item) - (38 if (item == item.upper()) else 96)

print(points)

File: day-3/challenge1.rs
use std::fs;

fn main() {
  let mut points: u64 = 0;
  let rucksacks = fs::read_to_string("input").expect("Unable to read file");

  for rucksack in rucksacks.lines() {
    let component1 = &rucksack[0..rucksack.chars().count() / 2];
    let component2 = &rucksack[rucksack.chars().count() / 2..];

    let result = component1.chars().find(|&x| component2.contains(x));

    if result.unwrap() == result.unwrap().to_uppercase().collect::<Vec<_>>()[0] {
      points += u64::from(result.unwrap() as u8) - 38;
    }
    else {
      points += u64::from(result.unwrap() as u8) - 96;
    }
  }

  println!("{}", points);
}
